appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/interceptors/InterceptorManager.java

```diff
@@ -594,23 +596,22 @@ private void initEjbCallbackIndices()
         callbackChain = new CallbackChainImpl[size];
         for (CallbackType eventType : CallbackType.values()) {
             int index = eventType.ordinal();
-            CallbackInterceptor[] interceptors = (CallbackInterceptor[])
-                    callbacks[index].toArray(new CallbackInterceptor[callbacks[index].size()]);
+            CallbackInterceptor[] interceptors = callbacks[index].toArray(new CallbackInterceptor[callbacks[index].size()]);
```

Comment:
The value on the index position is resolved twice, JIT will probably simplify it, but it could be simplified directly ... also then you would not need the index variable. 
Blah ... array of arraylist, who invented this? :D ... nevermind ...

Reply:
There is a lot of more cleaning possible. At some point I tried to keep this focused on generics and warnings I came across while doing that.